build_tools/update_environments_and_lock_files.py

```diff
@@ -89,9 +89,11 @@ def remove_from(alist, to_remove):
     return [each for each in alist if each not in to_remove]
 
 
-conda_build_metadata_list = [
+build_metadata_list = [
```

Comment:
How about making the entries/dicts dataclasses?

Reply:
Why not, but I would say this can be a separate PR, not necessarily by me :wink:. We could also go more OOP if we think this is useful, e.g. have `CondaBuild`, `PipBuild` and methods on them to generate environment and lock files.